scrapy_zyte_api/handler.py

```diff
@@ -21,11 +26,485 @@
 logger = logging.getLogger(__name__)
 
 
+_DEFAULT_API_PARAMS = {
+    "browserHtml": False,
+    "screenshot": False,
+}
+
+
+def _iter_headers(
+    *,
+    api_params: Dict[str, Any],
+    request: Request,
+    parameter: str,
+):
+    headers = api_params.get(parameter)
+    if headers not in (None, True):
+        logger.warning(
+            f"Request {request} defines the Zyte API {parameter} parameter, "
+            f"overriding Request.headers. Use Request.headers instead."
+        )
+        return
+    if not request.headers:
+        return
+    for k, v in request.headers.items():
+        if not v:
+            continue
+        decoded_v = b",".join(v).decode()
+        lowercase_k = k.strip().lower()
+        yield k, lowercase_k, decoded_v
+
+
+def _map_custom_http_request_headers(
```

Comment:
What do you think about moving automap-related utilities to a separate module? 

I was thinking about a function like this (I think you have something like this already):

```py
def get_zyte_api_request(request: scrapy.Request, zyte_automap: dict) -> dict[str, Any]:
    """ Get a scrapy.Request, as well as automap params (already merged with default params). 
    Return parameters for Zyte API request. """
```

Reply:
I am moving `_get_api_params` into its own module, but I am not sure if this is what you had in mind.